### PR TITLE
Used go-gh table printer for listing caches

### PR DESCRIPTION
**Pull request by t-dedah**
### What are you trying to accomplish?

Removing our printing logic with go-gh table printer

### Anything you want to highlight for special attention from reviewers?
No

### CODE REVIEW

**Review by kotewar — APPROVED** [review]
LGTM

**Comment by kotewar** [comment]
Do we release v1.0.1 with this table printer?

**Comment by t-dedah** [comment]
yes we should release a new version but we might pick up other minor changes so finalising those with Aparna.

File: internal/utils.go [hunk 1]
```diff
@@ -81,23 +76,6 @@ func lastAccessedTime(lastAccessedAt string) string {
 	return fmt.Sprintf(" %s", lastAccessed.FromNow())
```
Comment (bishal-pdMSFT) [inline internal/utils.go:76]: Dio we need this space `" %s"` now?

File: internal/utils.go [hunk 2]
```diff
@@ -51,17 +44,19 @@ func FormatCacheSize(size_in_bytes float64) string {
...
+		tp.AddField(FormatCacheSize(cache.SizeInBytes))
+		tp.AddField(cache.Ref)
+		tp.AddField(lastAccessedTime(cache.LastAccessedAt))
+		tp.EndRow()
 	}
+
+	_ = tp.Render()
 }
 
 func PrettyPrintTrimmedCacheList(caches []types.ActionsCache) {
```
Comment (bishal-pdMSFT) [inline internal/utils.go:62]: Is this method still used?

Comment (bishal-pdMSFT) [reply internal/utils.go:62]: If not then probably CI should have thrown an error about unused code. Can we enable that if go build allows this feature?

Comment (bishal-pdMSFT) [reply internal/utils.go:62]: Ah yes

**Comment by bishal-pdMSFT** [comment]
@t-dedah can you please paste some screenshots at different shell window sizes to help visualize. We can also add @N-Usha and @heyamie to check those.

File: internal/utils.go (hunk 1 above)
Comment (t-dedah) [reply internal/utils.go:76]: Not needed anymore, removed

**Comment by t-dedah** [comment]
Hi @bishal-pdMSFT @heyamie @N-Usha we have removed our printing logic and updated the code to reuse go-gh printer which is based on gh cli. Screenshots with latest changes:-

## Full width for the terminal

Screenshot 2022-08-09 at 4 21 27 PM

## Resizing the terminal with last printer result

Screenshot 2022-08-09 at 4 21 51 PM

## Printed after reducing the size of the window

Screenshot 2022-08-09 at 4 22 05 PM

## Further reduction
Screenshot 2022-08-09 at 4 22 23 PM

**Comment by bishal-pdMSFT** [comment]
Overall looks great. I am a bit worried about the key getting truncated a bit more larger terminal size than probably it was earlier. Basically we are still not handling #21

**Comment by bishal-pdMSFT** [comment]
> Overall looks great. I am a bit worried about the key getting truncated a bit more larger terminal size than probably it was earlier. Basically we are still not handling #21

As per discussion with @t-dedah the idea here is to enable full key visualization by using `--json` option. That will be a separate PR.

**Review by bishal-pdMSFT — APPROVED** [review]

**Pull request merged**